### ISSUE TITLE
Can't connect to self-hosted server with self-signed certificate from android app

### ISSUE THREAD

**Issue by cldtech** [RocketChat/Rocket.Chat#16612]
### Description:
Can't connect to self-hosted server with self-signed certificate from android app.
I did also tried to manually add the certificate in android.

### Steps to reproduce:
1. install rocketchat-server snap
2. enable https
3. connect with the android app

### Expected behavior:
Should ask if accept un-trusted authority. Just like the desktop app.

### Server Setup Information:
- Version of Rocket.Chat Server: 2.4.9
- Operating System: Ubuntu 18.04
- Deployment Method: snap
- NodeJS Version: v8.17.0
- MongoDB Version: 3.6.14

### Client Setup Information
- Desktop App or Browser Version: 4.3.1
- Operating System: Android

![Screenshot_2020-02-15-20-05-50](https://user-images.githubusercontent.com/52659208/74597570-8a2b4d00-502f-11ea-9d9a-20fab5243644.png)

**Comment by KrisJohnstone** [comment]
To clarify you:
- Have configured a reverse proxy ala:
https://rocket.chat/docs/installation/manual-installation/configuring-ssl-reverse-proxy/
- Used https when attempting to connect.

**Comment by cldtech** [comment]
Yes, the server part is fine, the desktop client work perfectly. I assume because it has the possibility to prompt an modal asking to accepting the self-signed certificate.

**Comment by petrerosioru** [comment]
I've also tried to install my self-signed certificate on my phone by using this guide: https://support.google.com/pixelphone/answer/2844832?hl=en
No luck, unfortunately.

I've installed quite a lot of reverse proxy servers, even with nginx and apache2, also to support websockets, so I think I know what I'm doing and what to test. I've even turned proxy_buffering, and still nothing.

At this point, I'm pretty sure that it's not a problem with the proxy server nor with the certificates.

I am open for workarounds, because my team is unable to get notifications on their Android phones.

Thank you!
